### PR TITLE
tools: make simulator work in large cluster

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #xxx

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

File: tools/pd-simulator/simulator/client.go [hunk 1]
```diff
@@ -378,8 +352,23 @@ func (c *client) StoreHeartbeat(ctx context.Context, stats *pdpb.StoreStats) err
...
+		Region:          region.GetMeta(),
+		Leader:          region.GetLeader(),
+		DownPeers:       region.GetDownPeers(),
+		PendingPeers:    region.GetPendingPeers(),
+		BytesWritten:    region.GetBytesWritten(),
+		BytesRead:       region.GetBytesRead(),
+		ApproximateSize: uint64(region.GetApproximateSize()),
+		ApproximateKeys: uint64(region.GetApproximateKeys()),
+	}
+	err := c.stream.Send(request)
```
Comment (lilinghai) [inline tools/pd-simulator/simulator/client.go:367]: maybe the RegionHeartbeat func doesn't need return the data of error type

**Comment by nolouch** [comment]
cc @HunDunDM

File: tools/pd-simulator/simulator/cases/balance_leader.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 func newBalanceLeader() *Case {
 	var simCase Case
 
-	storeNum, regionNum := getStoreNum(), getRegionNum()
+	storeNum, regionNum := 100, 2000000
```
[the remark this answers is not in the document]
Comment (nolouch) [reply tools/pd-simulator/simulator/cases/balance_leader.go:29]: it's better be configurable.

**Comment by rleungx** [comment]
Close it since it's stale.

**Pull request closed without merging**